sfcasts/crud-controller.md

```diff
@@ -1,171 +1,90 @@
-controller to be the real controller. In this case, to be question crowd controller
-index, you can actually see this down here on the web Depot toolbar. If you hover
-over these a admin at admin, tell this tells you that the route name is admin. So the
-route name is actually this route right here, but the controller is question crud
-controller, call on call an index. So there's some route magic happening, but
-ultimately the methods in your crud controller are the ones that are called. So it's
-the index method in this abstract crud controller down here. This is actually the
-real roller for the page. This is important for security, just because if we had only
-put the route and the route attribute above index and not added and not added the
-access control, that wouldn't have been enough
+# CRUD Controller
+
+The true reason you use easy admin is for its CRUD controllers. Each CRUD controller will give us a rich set of pages to create, read, update, and delete a single entity. This is where EasyAdmin shines, and the next few minutes are going to be critically important to understanding how EasyAdmin works.
+
+We have four entities, so let's generate a CRUD controller. We'll do that for `Question` first. Find a terminal and run:
+
+```
+symfony console make:admin:crud
+```
+
+As you can see, it recognizes our four entities. I'll hit 1 for `App/Entity/Question`. Let this generate into the default directory and with default namespace.
+
+Sweet! This did exactly one thing. It created a new `QuestionCrudController` file. Let's go open that up.
+
+Before we look too deeply into this, head over to the admin page and refresh to see... absolutely no difference. We *do* have a new `QuestionCrudController`, but these CRUD controllers are kind of useless until you link to them from a dashboard. So, back over on a `DashboardController`, down at the bottom... `yield MenuItem`, but instead of link to dashboard, there's a number of other things that you can link to. We want to `LinkToCrud`. We're gonna pass this a couple things: The label... so `questions`... and an icon, `fa fa-question-circle`. That's a FontAwesome icon. And then, most importantly, the entity's class
+name. So `Question::class`. Behind the scenes, EasyAdmin's going to recognize that there's only one CRUD dashboard for this entity, which is `QuestionCrudController`, and it will know to use this. And yes, in theory, you can have multiple CRUD controllers per entity. That's something we'll talk about later.
+
+Okay, go refresh to reveal our new link, click and... whoa! This is super cool! We have an `isApproved` slider, which saves automatically. We also have a search bar on top... and sortable columns to help us find whatever we're looking for.
+
+We can edit, delete the edit, and it even has a cool calendar. We are absolutely loaded with awesome features. It's pretty cool! So let's repeat this for our other three controllers.
```

Comment:
I just noticed the use of "super cool" in the paragraph before, and also describing the calendar feature here, so perhaps I should substitute another excited word for "pretty cool" so it doesn't sound repetitive.